engines/agi/agi.cpp

```diff
@@ -520,30 +274,419 @@ AgiBase::~AgiBase() {
 	delete _rnd;
 
 	delete _sound;
+
+	if (_fontDataAllocated) {
+		free(_fontDataAllocated);
+	}
 }
 
 void AgiBase::initRenderMode() {
-	_renderMode = Common::kRenderEGA;
+	Common::Platform platform = Common::parsePlatform(ConfMan.get("platform"));
+	Common::RenderMode configRenderMode = Common::parseRenderMode(ConfMan.get("render_mode").c_str());
+
+	// Default to EGA PC rendering
+	_renderMode = RENDERMODE_EGA;
+
+	switch (platform) {
+	case Common::kPlatformDOS:
+		switch (configRenderMode) {
+		case Common::kRenderCGA:
+			_renderMode = RENDERMODE_CGA;
+			break;
+		// Hercules is not supported atm
+		//case Common::kRenderHercA:
+		//case Common::kRenderHercG:
+		//	_renderMode = RENDERMODE_HERCULES;
+		//	break;
+		default:
+			break;
+		}
+		break;
+	case Common::kPlatformAmiga:
+		_renderMode = RENDERMODE_AMIGA;
+		break;
+	case Common::kPlatformApple2GS:
+		_renderMode = RENDERMODE_APPLE_II_GS;
+		break;
+	case Common::kPlatformAtariST:
+		_renderMode = RENDERMODE_ATARI_ST;
+		break;
+	default:
+		break;
+	}
 
-	if (ConfMan.hasKey("platform")) {
-		Common::Platform platform = Common::parsePlatform(ConfMan.get("platform"));
-		_renderMode = (platform == Common::kPlatformAmiga) ? Common::kRenderAmiga : Common::kRenderEGA;
+	// If render mode is explicitly set, force rendermode
+	switch (configRenderMode) {
+	case Common::kRenderAmiga:
+		_renderMode = RENDERMODE_AMIGA;
+		break;
+	case Common::kRenderApple2GS:
+		_renderMode = RENDERMODE_APPLE_II_GS;
+		break;
+	case Common::kRenderAtariST:
+		_renderMode = RENDERMODE_ATARI_ST;
+		break;
+	default:
+		break;
 	}
 
-	if (ConfMan.hasKey("render_mode")) {
-		Common::RenderMode tmpMode = Common::parseRenderMode(ConfMan.get("render_mode").c_str());
-		if (tmpMode != Common::kRenderDefault)
-			_renderMode = tmpMode;
+	if (getFeatures() & (GF_AGI256 | GF_AGI256_2)) {
+		// If current game is AGI256, switch (force) to VGA render mode
+		_renderMode = RENDERMODE_VGA;
 	}
 }
 
+void AgiBase::initFont() {
+	// We are currently using the custom font for all fanmade games
```

Comment:
I recommend to move all these font-related methods to font.cpp file (leaving them in AgiBase class, though).